huaweicloud/data_source_huaweicloud_nat_gateway_v2_test.go

```diff
@@ -0,0 +1,93 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccNatGatewayDataSource_basic(t *testing.T) {
+	natgateway := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNatGatewayV2DataSource_basic(natgateway),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNatGatewayV2DataSourceID("data.huaweicloud_nat_gateway.nat_by_name"),
+					testAccCheckNatGatewayV2DataSourceID("data.huaweicloud_nat_gateway.nat_by_id"),
+					resource.TestCheckResourceAttr(
+						"data.huaweicloud_nat_gateway.nat_by_name", "name", natgateway),
+					resource.TestCheckResourceAttr(
+						"data.huaweicloud_nat_gateway.nat_by_id", "name", natgateway),
+					resource.TestCheckResourceAttr(
+						"data.huaweicloud_nat_gateway.nat_by_name", "admin_state_up", "true"),
+					resource.TestCheckResourceAttr(
+						"data.huaweicloud_nat_gateway.nat_by_id", "admin_state_up", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckNatGatewayV2DataSourceID(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Can't find natgateway data source: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("NatGateway data source ID not set")
+		}
+
+		return nil
+	}
+}
+
+func testAccNatGatewayV2DataSource_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_networking_router_v2" "router_1" {
+  name = "router_1"
+  admin_state_up = "true"
+}
+
+resource "huaweicloud_networking_network_v2" "network_1" {
+  name = "network_1"
+  admin_state_up = "true"
+}
+
+resource "huaweicloud_networking_subnet_v2" "subnet_1" {
+  cidr = "192.168.199.0/24"
+  ip_version = 4
+  network_id = "${huaweicloud_networking_network_v2.network_1.id}"
+}
+
+resource "huaweicloud_networking_router_interface_v2" "int_1" {
+  subnet_id = "${huaweicloud_networking_subnet_v2.subnet_1.id}"
+  router_id = "${huaweicloud_networking_router_v2.router_1.id}"
+}
```

Comment:
these networking resources are all deprecated, use vpc and vpc_subnet resources instead.